src/test/scala/mesosphere/marathon/api/akkahttp/v2/AppsControllerTest.scala

```diff
@@ -0,0 +1,1433 @@
+import scala.util.Try
+import scala.util.control.NonFatal
+
+class AppsControllerTest extends UnitTest with GroupCreation with ScalatestRouteTest {
+
+  case class Fixture(
+      clock: SettableClock = new SettableClock(),
+      auth: TestAuthFixture = new TestAuthFixture {
+        override val UnauthorizedStatus = 403
+        override val NotAuthenticatedStatus = 401
+      },
+      service: MarathonSchedulerService = mock[MarathonSchedulerService],
+      appInfoService: AppInfoService = mock[AppInfoService],
+      healthCheckManager: HealthCheckManager = mock[HealthCheckManager],
+      instanceTracker: InstanceTracker = mock[InstanceTracker],
+      taskKiller: TaskKiller = mock[TaskKiller],
+      configArgs: Seq[String] = Seq("--enable_features", "external_volumes"),
+      groupManager: GroupManager = mock[GroupManager]) {
+    val config: AllConf = AllConf.withTestConfig(configArgs: _*)
+
+    implicit val authenticator: Authenticator = auth.auth
+    implicit val authorizer: Authorizer = auth.auth
+
+    val normalizationConfig = AppNormalization.Configuration(config.defaultNetworkName.get, config.mesosBridgeName())
+    implicit lazy val appDefinitionValidator = AppDefinition.validAppDefinition(config.availableFeatures)(PluginManager.None)
+    implicit lazy val validateCanonicalAppUpdateAPI = AppValidation.validateCanonicalAppUpdateAPI(config.availableFeatures, () => config.defaultNetworkName.get)
+
+    implicit val validateAndNormalizeApp: Normalization[raml.App] =
+      AppHelpers.appNormalization(config.availableFeatures, normalizationConfig)(AppNormalization.withCanonizedIds())
+
+    implicit val validateAndNormalizeAppUpdate: Normalization[raml.AppUpdate] =
+      AppHelpers.appUpdateNormalization(config.availableFeatures, normalizationConfig)(AppNormalization.withCanonizedIds())
+
+    implicit val electionService: ElectionService = mock[ElectionService]
+
+    when(electionService.isLeader).thenReturn(true)
+
+    val appsController = new AppsController(
+      clock = clock,
+      eventBus = system.eventStream,
+      marathonSchedulerService = service,
+      appInfoService = appInfoService,
+      healthCheckManager = healthCheckManager,
+      instanceTracker = instanceTracker,
+      taskKiller = taskKiller,
+      config = config,
+      groupManager = groupManager,
+      pluginManager = PluginManager.None
+    )
+
+    implicit val rejectionHandler: RejectionHandler = AkkaHttpMarathonService.rejectionHandler
+    val route: Route = Route.seal(appsController.route)
+
+    def normalize(app: App): App = {
+      val migrated = AppNormalization.forDeprecated(normalizationConfig).normalized(app)
+      AppNormalization(normalizationConfig).normalized(migrated)
+    }
+
+    def normalizeAndConvert(app: App): AppDefinition = {
+      val normalized = normalize(app)
+      Raml.fromRaml(normalized)
+    }
+
+    def prepareApp(app: App, groupManager: GroupManager): (Array[Byte], DeploymentPlan) = {
```

Comment:
If I understand it correctly, it will update mocked group manager with appropriate values for the given app.

Reply:
Hm, this seems too much for a test to me but I understand that you've just copied it. Since I cannot think of a better solution for now I'm fine with it.